teacher.rb

```diff
@@ -0,0 +1,13 @@
+require './person'
+
+# Represents a teacher with a specific specialization.
+class Teacher < Person
+  def initialize(age, specialization, name = 'unknown', parent_permission: true)
+    super(age, name, parent_permission)
```

Comment:
- **[OPTIONAL]** Since the `parent_permission` uses colon notation, its use is somewhat different from what we are used to. In the super parameters, the parent_permission should be this.

  ```
  parent_permission: parent_permission
  ```